File: lena-config/src/main/java/org/riversoforion/lena/config/resolvers/EnvironmentValueResolver.java
/*
 * Copyright (c) 2024. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers;

import org.riversoforion.lena.config.ValueResolver;

import java.util.Optional;

public class EnvironmentValueResolver implements ValueResolver {

    @Override
    public Optional<String> resolveValue(String name) {

        return Optional.ofNullable(System.getenv(name));
    }
}

File: lena-config/src/commonMain/kotlin/org/riversoforion/lena/config/resolvers/EnvironmentValueResolver.kt
/*
 * Copyright (c) 2024-2026. Eric McIntyre / Rivers of Orion
 */
package org.riversoforion.lena.config.resolvers

import org.riversoforion.lena.config.ValueResolver
import org.riversoforion.lena.config.internal.platformGetEnv

/** Looks up a value from the process environment using [platformGetEnv]. */
public class EnvironmentValueResolver : ValueResolver {
    override fun resolveValue(name: String): String? = platformGetEnv(name)
}
